### DISCUSSION TITLE
How to modify allure trace in pytest

### DISCUSSION THREAD

**Discussion by lozik4** [allure-framework/allure2#2272] [Questions & Support] [answered]
I have function
```
def get_element(self, locator: str) -> WebElement:
        try:
            return self.driver.find_element(self.loc, locator)
        except Exception as e:
            stacktrace = None
            msg = f"__ERROR__: {e.__class__.__name__} No such element {locator}"
            raise WebDriverException(msg=msg, stacktrace=stacktrace)
```
           
 when work exception in allure result I see trace NoSuchElementException
 It's very strange. Because I disable the trace in my exception. Haw, I can not log a trace in pytest? 
<img width="1664" alt="Screenshot 2024-01-05 at 19 52 51" src="https://github.com/allure-framework/allure2/assets/42421616/c3c3dfdb-0826-4a7d-96de-d6f07374f87e">

**Comment by delatrie** [comment, marked as answer]
The original exception (which still has the `stacktrace`) is reported due to the [exception chaining](https://peps.python.org/pep-3134/) feature of Python 3. Use the following syntax to suppress the chaining explicitly:

```python
raise WebDriverException(msg=msg, stacktrace=stacktrace) from None
```

Clearing the `stacktrace` of the original exception should work too since it's just a custom attribute managed by Selenium during exception formatting:

```python
try:
    ...
except Exception as e:
    e.stacktrace = None
    raise
```

**Reply by lozik4** [reply]
Thanks, it's work.
